### ISSUE TITLE
Enhancement: Support for Browser environments like WebPack/Browserify/eslint-linter-browserify

### ISSUE THREAD

**Issue by as1605** [typescript-eslint/typescript-eslint#7123]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Adding support for running this parser alongside [eslint-linter-browserify](https://www.npmjs.com/package/eslint-linter-browserify)

This patch is for projects which wish to run this on browser environments using Webpack/Browserify, for example to make an online editor and linter.

I have enforced Isolated Program environment, and removed all references to `fs` and `path` modules which require a NodeJS environment to run

This should fix requirements like https://github.com/UziTech/eslint-linter-browserify/issues/213

### Fail

```typescript
configurations which load files or modules/plugins from directory paths or watch projects
```

### Pass

```typescript
TypeScript isolated files. Configurations have to be provided as functions
```

### Additional Info

Here is the diff (I have removed the files which are completely deleted from the diff due to max body size limit)
```diff
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.d.ts b/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.d.ts
deleted file mode 100644
index 1845702..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.d.ts
+++ /dev/null
@@ -1,10 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.d.ts.map b/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.d.ts.map
deleted file mode 100644
index dbfae70..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.d.ts.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.js b/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.js
deleted file mode 100644
index 5ad7a1d..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.js
+++ /dev/null
@@ -1,24 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.js.map b/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.js.map
deleted file mode 100644
index 951ca84..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/clear-caches.js.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.d.ts b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.d.ts
deleted file mode 100644
index 0165846..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.d.ts
+++ /dev/null
@@ -1,9 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.d.ts.map b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.d.ts.map
deleted file mode 100644
index d96dc85..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.d.ts.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.js
deleted file mode 100644
index a51cf28..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.js
+++ /dev/null
@@ -1,65 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.js.map b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.js.map
deleted file mode 100644
index a6ac9ec..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createDefaultProgram.js.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.d.ts b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.d.ts
deleted file mode 100644
index 35fdf3d..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.d.ts
+++ /dev/null
@@ -1,9 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.d.ts.map b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.d.ts.map
deleted file mode 100644
index 7b7f9df..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.d.ts.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js
deleted file mode 100644
index 59c382c..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js
+++ /dev/null
@@ -1,102 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js.map b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js.map
deleted file mode 100644
index 438aac1..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js.map
+++ /dev/null
@@ -1 +0,0 @@
-{"version":3,"file":"createProjectProgram.js","sourceRoot":"","sources":["../../src/create-program/createProjectProgram.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,kDAA0B;AAC1B,gDAAwB;AACxB,+CAAiC;AAEjC,8CAA6C;AAE7C,yDAAsD;AACtD,+EAA4E;AAE5E,qCAA6C;AAE7C,MAAM,GAAG,GAAG,IAAA,eAAK,EAAC,0DAA0D,CAAC,CAAC;AAE9E,MAAM,6BAA6B,GAAG;IACpC,EAAE,CAAC,SAAS,CAAC,EAAE;IACf,EAAE,CAAC,SAAS,CAAC,GAAG;IAChB,EAAE,CAAC,SAAS,CAAC,EAAE;IACf,EAAE,CAAC,SAAS,CAAC,GAAG;IAChB,EAAE,CAAC,SAAS,CAAC,GAAG;IAChB,EAAE,CAAC,SAAS,CAAC,GAAG;IAChB,EAAE,CAAC,SAAS,CAAC,GAAG;IAChB,EAAE,CAAC,SAAS,CAAC,GAAG;CACI,CAAC;AAEvB;;;GAGG;AACH,SAAS,oBAAoB,CAC3B,aAA4B;IAE5B,GAAG,CAAC,kCAAkC,EAAE,aAAa,CAAC,QAAQ,CAAC,CAAC;IAEhE,MAAM,mBAAmB,GAAG,IAAA,yDAA2B,EAAC,aAAa,CAAC,CAAC;IACvE,MAAM,aAAa,GAAG,IAAA,yBAAY,EAAC,mBAAmB,EAAE,cAAc,CAAC,EAAE,CACvE,IAAA,0BAAiB,EAAC,cAAc,EAAE,aAAa,CAAC,CACjD,CAAC;IAEF,0FAA0F;IAC1F,IAAI,aAAa,IAAI,aAAa,CAAC,oBAAoB,EAAE;QACvD,OAAO,aAAa,CAAC;KACtB;IAED,MAAM,uBAAuB,GAAG,CAAC,WAAmB,EAAU,EAAE,CAC9D,IAAA,mCAAgB,EAAC,WAAW,EAAE,aAAa,CAAC,eAAe,CAAC,CAAC;IAE/D,MAAM,iBAAiB,GAAG,IAAA,mCAAgB,EACxC,aAAa,CAAC,QAAQ,EACtB,aAAa,CAAC,eAAe,CAC9B,CAAC;IACF,MAAM,gBAAgB,GAAG,aAAa,CAAC,QAAQ,CAAC,GAAG,CAAC,uBAAuB,CAAC,CAAC;IAC7E,MAAM,iBAAiB,GACrB,gBAAgB,CAAC,MAAM,KAAK,CAAC;QAC3B,CAAC,CAAC,gBAAgB,CAAC,CAAC,CAAC;QACrB,CAAC,CAAC,KAAK,gBAAgB,CAAC,GAAG,CAAC,OAAO,CAAC,EAAE,CAAC,KAAK,OAAO,EAAE,CAAC,CAAC,IAAI,CAAC,IAAI,CAAC,EAAE,CAAC;IACxE,MAAM,UAAU,GAAG;QACjB,qCAAqC,iBAAiB,uCAAuC,iBAAiB,EAAE;KACjH,CAAC;IACF,IAAI,iBAAiB,GAAG,KAAK,CAAC;IAE9B,MAAM,mBAAmB,GAAG,aAAa,CAAC,mBAAmB,IAAI,EAAE,CAAC;IAEpE,mBAAmB,CAAC,OAAO,CAAC,cAAc,CAAC,EAAE;QAC3C,IAAI,CAAC,cAAc,CAAC,UAAU,CAAC,GAAG,CAAC,EAAE;YACnC,UAAU,CAAC,IAAI,CACb,gCAAgC,cAAc,uFAAuF,cAAc,KAAK,CACzJ,CAAC;SACH;QACD,IAAI,6BAA6B,CAAC,QAAQ,CAAC,cAAc,CAAC,EAAE;YAC1D,UAAU,CAAC,IAAI,CACb,8CAA8C,cAAc,uHAAuH,CACpL,CAAC;SACH;IACH,CAAC,CAAC,CAAC;IAEH,MAAM,aAAa,GAAG,cAAI,CAAC,OAAO,CAAC,aAAa,CAAC,QAAQ,CAAC,CAAC;IAC3D,IAAI,CAAC,6BAA6B,CAAC,QAAQ,CAAC,aAAa,CAAC,EAAE;QAC1D,MAAM,cAAc,GAAG,iCAAiC,aAAa,qBAAqB,CAAC;QAC3F,IAAI,mBAAmB,CAAC,MAAM,GAAG,CAAC,EAAE;YAClC,IAAI,CAAC,mBAAmB,CAAC,QAAQ,CAAC,aAAa,CAAC,EAAE;gBAChD,UAAU,CAAC,IAAI,CACb,GAAG,cAAc,8EAA8E,CAChG,CAAC;gBACF,iBAAiB,GAAG,IAAI,CAAC;aAC1B;SACF;aAAM;YACL,UAAU,CAAC,IAAI,CACb,GAAG,cAAc,wEAAwE,CAC1F,CAAC;YACF,iBAAiB,GAAG,IAAI,CAAC;SAC1B;KACF;IAED,IAAI,CAAC,iBAAiB,EAAE;QACtB,MAAM,CAAC,mBAAmB,EAAE,mBAAmB,CAAC,GAC9C,aAAa,CAAC,QAAQ,CAAC,MAAM,KAAK,CAAC;YACjC,CAAC,CAAC,CAAC,wBAAwB,EAAE,eAAe,CAAC;YAC7C,CAAC,CAAC,CAAC,yBAAyB,EAAE,wBAAwB,CAAC,CAAC;QAC5D,UAAU,CAAC,IAAI,CACb,YAAY,mBAAmB,6BAA6B,EAC5D,mEAAmE,EACnE,YAAY,mBAAmB,uBAAuB,EACtD,8FAA8F,EAC9F,oOAAoO,CACrO,CAAC;KACH;IAED,MAAM,IAAI,KAAK,CAAC,UAAU,CAAC,IAAI,CAAC,IAAI,CAAC,CAAC,CAAC;AACzC,CAAC;AAEQ,oDAAoB"}
\ No newline at end of file
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/describeFilePath.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/describeFilePath.js
index ba35361..91ee5be 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/describeFilePath.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/describeFilePath.js
@@ -4,29 +4,30 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.describeFilePath = void 0;
-const path_1 = __importDefault(require("path"));
+// const path_1 = __importDefault(require("path"));
 function describeFilePath(filePath, tsconfigRootDir) {
-    // If the TSConfig root dir is a parent of the filePath, use
-    // `<tsconfigRootDir>` as a prefix for the path.
-    const relative = path_1.default.relative(tsconfigRootDir, filePath);
-    if (relative && !relative.startsWith('..') && !path_1.default.isAbsolute(relative)) {
-        return `<tsconfigRootDir>/${relative}`;
-    }
-    // Root-like Mac/Linux (~/*, ~*) or Windows (C:/*, /) paths that aren't
-    // relative to the TSConfig root dir should be fully described.
-    // This avoids strings like <tsconfigRootDir>/../../../../repo/file.ts.
-    // https://github.com/typescript-eslint/typescript-eslint/issues/6289
-    if (/^[(\w+:)\\/~]/.test(filePath)) {
-        return filePath;
-    }
-    // Similarly, if the relative path would contain a lot of ../.., then
-    // ignore it and print the file path directly.
-    if (/\.\.[/\\]\.\./.test(relative)) {
-        return filePath;
-    }
-    // Lastly, since we've eliminated all special cases, we know the cleanest
-    // path to print is probably the prefixed relative one.
-    return `<tsconfigRootDir>/${relative}`;
+    return filePath;
+    // // If the TSConfig root dir is a parent of the filePath, use
+    // // `<tsconfigRootDir>` as a prefix for the path.
+    // const relative = path_1.default.relative(tsconfigRootDir, filePath);
+    // if (relative && !relative.startsWith('..') && !path_1.default.isAbsolute(relative)) {
+    //     return `<tsconfigRootDir>/${relative}`;
+    // }
+    // // Root-like Mac/Linux (~/*, ~*) or Windows (C:/*, /) paths that aren't
+    // // relative to the TSConfig root dir should be fully described.
+    // // This avoids strings like <tsconfigRootDir>/../../../../repo/file.ts.
+    // // https://github.com/typescript-eslint/typescript-eslint/issues/6289
+    // if (/^[(\w+:)\\/~]/.test(filePath)) {
+    //     return filePath;
+    // }
+    // // Similarly, if the relative path would contain a lot of ../.., then
+    // // ignore it and print the file path directly.
+    // if (/\.\.[/\\]\.\./.test(relative)) {
+    //     return filePath;
+    // }
+    // // Lastly, since we've eliminated all special cases, we know the cleanest
+    // // path to print is probably the prefixed relative one.
+    // return `<tsconfigRootDir>/${relative}`;
 }
 exports.describeFilePath = describeFilePath;
 //# sourceMappingURL=describeFilePath.js.map
\ No newline at end of file
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getScriptKind.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getScriptKind.js
index 964439b..d1ea6fa 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getScriptKind.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getScriptKind.js
@@ -27,10 +27,12 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.getLanguageVariant = exports.getScriptKind = void 0;
-const path_1 = __importDefault(require("path"));
+// const path_1 = __importDefault(require("path"));
 const ts = __importStar(require("typescript"));
 function getScriptKind(filePath, jsx) {
-    const extension = path_1.default.extname(filePath).toLowerCase();
+    const arr = filePath.toLowerCase().split('.');
+    const extension = arr[arr.length - 1];
+    // path_1.default.extname(filePath).toLowerCase();
     // note - we only respect the user's jsx setting for unknown extensions
     // this is so that we always match TS's internal script kind logic, preventing
     // weird errors due to a mismatch.
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.d.ts b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.d.ts
deleted file mode 100644
index 621d9bd..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.d.ts
+++ /dev/null
@@ -1,15 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.d.ts.map b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.d.ts.map
deleted file mode 100644
index 07ea210..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.d.ts.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js
deleted file mode 100644
index 8b95cea..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js
+++ /dev/null
@@ -1,409 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js.map b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js.map
deleted file mode 100644
index 8647bfe..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/getWatchProgramsForProjects.js.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/shared.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/shared.js
index 89d5fbf..70a0c09 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/shared.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/shared.js
@@ -27,7 +27,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.getModuleResolver = exports.getAstFromProgram = exports.getCanonicalFileName = exports.ensureAbsolutePath = exports.createHash = exports.createDefaultCompilerOptionsFromExtra = exports.canonicalDirname = exports.CORE_COMPILER_OPTIONS = void 0;
-const path_1 = __importDefault(require("path"));
+// const path_1 = __importDefault(require("path"));
 const ts = __importStar(require("typescript"));
 /**
  * Compiler options required to avoid critical functionality issues
@@ -57,35 +57,36 @@ const useCaseSensitiveFileNames = ts.sys !== undefined ? ts.sys.useCaseSensitive
 const correctPathCasing = useCaseSensitiveFileNames
     ? (filePath) => filePath
     : (filePath) => filePath.toLowerCase();
-function getCanonicalFileName(filePath) {
-    let normalized = path_1.default.normalize(filePath);
-    if (normalized.endsWith(path_1.default.sep)) {
-        normalized = normalized.slice(0, -1);
-    }
-    return correctPathCasing(normalized);
-}
-exports.getCanonicalFileName = getCanonicalFileName;
-function ensureAbsolutePath(p, tsconfigRootDir) {
-    return path_1.default.isAbsolute(p)
-        ? p
-        : path_1.default.join(tsconfigRootDir || process.cwd(), p);
-}
-exports.ensureAbsolutePath = ensureAbsolutePath;
-function canonicalDirname(p) {
-    return path_1.default.dirname(p);
-}
-exports.canonicalDirname = canonicalDirname;
+// function getCanonicalFileName(filePath) {
+//     let normalized = path_1.default.normalize(filePath);
+//     if (normalized.endsWith(path_1.default.sep)) {
+//         normalized = normalized.slice(0, -1);
+//     }
+//     return correctPathCasing(normalized);
+// }
+// exports.getCanonicalFileName = getCanonicalFileName;
+// function ensureAbsolutePath(p, tsconfigRootDir) {
+//     return path_1.default.isAbsolute(p)
+//         ? p
+//         : path_1.default.join(tsconfigRootDir || process.cwd(), p);
+// }
+// exports.ensureAbsolutePath = ensureAbsolutePath;
+// function canonicalDirname(p) {
+//     return path_1.default.dirname(p);
+// }
+// exports.canonicalDirname = canonicalDirname;
 const DEFINITION_EXTENSIONS = [
     ts.Extension.Dts,
     ts.Extension.Dcts,
     ts.Extension.Dmts,
 ];
 function getExtension(fileName) {
-    var _a;
+    var _a = "tsx";
     if (!fileName) {
         return null;
     }
-    return ((_a = DEFINITION_EXTENSIONS.find(definitionExt => fileName.endsWith(definitionExt))) !== null && _a !== void 0 ? _a : path_1.default.extname(fileName));
+    return ((_a = DEFINITION_EXTENSIONS.find(definitionExt => fileName.endsWith(definitionExt))) !== null && _a);
+    // _a !== void 0 ? _a : path_1.default.extname(fileName));
 }
 function getAstFromProgram(currentProgram, parseSettings) {
     const ast = currentProgram.getSourceFile(parseSettings.filePath);
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/useProvidedPrograms.js b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/useProvidedPrograms.js
index 5b291e9..b183817 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/create-program/useProvidedPrograms.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/create-program/useProvidedPrograms.js
@@ -28,8 +28,8 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.createProgramFromConfigFile = exports.useProvidedPrograms = void 0;
 const debug_1 = __importDefault(require("debug"));
-const fs = __importStar(require("fs"));
-const path = __importStar(require("path"));
+// const fs = __importStar(require("fs"));
+// const path = __importStar(require("path"));
 const ts = __importStar(require("typescript"));
 const shared_1 = require("./shared");
 const log = (0, debug_1.default)('typescript-eslint:typescript-estree:useProvidedProgram');
@@ -43,14 +43,14 @@ function useProvidedPrograms(programInstances, parseSettings) {
             break;
         }
     }
-    if (!astAndProgram) {
-        const relativeFilePath = path.relative(parseSettings.tsconfigRootDir || process.cwd(), parseSettings.filePath);
-        const errorLines = [
-            '"parserOptions.programs" has been provided for @typescript-eslint/parser.',
-            `The file was not found in any of the provided program instance(s): ${relativeFilePath}`,
-        ];
-        throw new Error(errorLines.join('\n'));
-    }
+    // if (!astAndProgram) {
+    //     const relativeFilePath = path.relative(parseSettings.tsconfigRootDir || process.cwd(), parseSettings.filePath);
+    //     const errorLines = [
+    //         '"parserOptions.programs" has been provided for @typescript-eslint/parser.',
+    //         `The file was not found in any of the provided program instance(s): ${relativeFilePath}`,
+    //     ];
+    //     throw new Error(errorLines.join('\n'));
+    // }
     astAndProgram.program.getTypeChecker(); // ensure parent pointers are set in source files
     return astAndProgram;
 }
@@ -61,28 +61,28 @@ exports.useProvidedPrograms = useProvidedPrograms;
  * @param configFile the path to the tsconfig.json file, relative to `projectDirectory`
  * @param projectDirectory the project directory to use as the CWD, defaults to `process.cwd()`
  */
-function createProgramFromConfigFile(configFile, projectDirectory) {
-    if (ts.sys === undefined) {
-        throw new Error('`createProgramFromConfigFile` is only supported in a Node-like environment.');
-    }
-    const parsed = ts.getParsedCommandLineOfConfigFile(configFile, shared_1.CORE_COMPILER_OPTIONS, {
-        onUnRecoverableConfigFileDiagnostic: diag => {
-            throw new Error(formatDiagnostics([diag])); // ensures that `parsed` is defined.
-        },
-        fileExists: fs.existsSync,
-        getCurrentDirectory: () => (projectDirectory && path.resolve(projectDirectory)) || process.cwd(),
-        readDirectory: ts.sys.readDirectory,
-        readFile: file => fs.readFileSync(file, 'utf-8'),
-        useCaseSensitiveFileNames: ts.sys.useCaseSensitiveFileNames,
-    });
-    const result = parsed; // parsed is not undefined, since we throw on failure.
-    if (result.errors.length) {
-        throw new Error(formatDiagnostics(result.errors));
-    }
-    const host = ts.createCompilerHost(result.options, true);
-    return ts.createProgram(result.fileNames, result.options, host);
-}
-exports.createProgramFromConfigFile = createProgramFromConfigFile;
+// function createProgramFromConfigFile(configFile, projectDirectory) {
+//     if (ts.sys === undefined) {
+//         throw new Error('`createProgramFromConfigFile` is only supported in a Node-like environment.');
+//     }
+//     const parsed = ts.getParsedCommandLineOfConfigFile(configFile, shared_1.CORE_COMPILER_OPTIONS, {
+//         onUnRecoverableConfigFileDiagnostic: diag => {
+//             throw new Error(formatDiagnostics([diag])); // ensures that `parsed` is defined.
+//         },
+//         fileExists: fs.existsSync,
+//         getCurrentDirectory: () => (projectDirectory && path.resolve(projectDirectory)) || process.cwd(),
+//         readDirectory: ts.sys.readDirectory,
+//         readFile: file => fs.readFileSync(file, 'utf-8'),
+//         useCaseSensitiveFileNames: ts.sys.useCaseSensitiveFileNames,
+//     });
+//     const result = parsed; // parsed is not undefined, since we throw on failure.
+//     if (result.errors.length) {
+//         throw new Error(formatDiagnostics(result.errors));
+//     }
+//     const host = ts.createCompilerHost(result.options, true);
+//     return ts.createProgram(result.fileNames, result.options, host);
+// }
+// exports.createProgramFromConfigFile = createProgramFromConfigFile;
 function formatDiagnostics(diagnostics) {
     return ts.formatDiagnostics(diagnostics, {
         getCanonicalFileName: f => f,
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/index.js b/node_modules/@typescript-eslint/typescript-estree/dist/index.js
index ca1e306..166d1f3 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/index.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/index.js
@@ -28,7 +28,7 @@ __exportStar(require("./create-program/getScriptKind"), exports);
 var version_check_1 = require("./version-check");
 Object.defineProperty(exports, "typescriptVersionIsAtLeast", { enumerable: true, get: function () { return version_check_1.typescriptVersionIsAtLeast; } });
 __exportStar(require("./getModifiers"), exports);
-__exportStar(require("./clear-caches"), exports);
+// __exportStar(require("./clear-caches"), exports);
 // re-export for backwards-compat
 var visitor_keys_1 = require("@typescript-eslint/visitor-keys");
 Object.defineProperty(exports, "visitorKeys", { enumerable: true, get: function () { return visitor_keys_1.visitorKeys; } });
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/createParseSettings.js b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/createParseSettings.js
index 772cf83..39da6cc 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/createParseSettings.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/createParseSettings.js
@@ -9,7 +9,7 @@ const shared_1 = require("../create-program/shared");
 const ExpiringCache_1 = require("./ExpiringCache");
 const getProjectConfigFiles_1 = require("./getProjectConfigFiles");
 const inferSingleRun_1 = require("./inferSingleRun");
-const resolveProjectList_1 = require("./resolveProjectList");
+// const resolveProjectList_1 = require("./resolveProjectList");
 const warnAboutTSVersion_1 = require("./warnAboutTSVersion");
 const log = (0, debug_1.default)('typescript-eslint:typescript-estree:parser:parseSettings:createParseSettings');
 let TSCONFIG_MATCH_CACHE;
@@ -36,9 +36,10 @@ function createParseSettings(code, options = {}) {
             options.extraFileExtensions.every(ext => typeof ext === 'string')
             ? options.extraFileExtensions
             : [],
-        filePath: (0, shared_1.ensureAbsolutePath)(typeof options.filePath === 'string' && options.filePath !== '<input>'
-            ? options.filePath
-            : getFileName(options.jsx), tsconfigRootDir),
+        filePath: options.filePath,
+        // (0, shared_1.ensureAbsolutePath)(typeof options.filePath === 'string' && options.filePath !== '<input>'
+        //     ? options.filePath
+        //     : getFileName(options.jsx), tsconfigRootDir),
         jsx: options.jsx === true,
         loc: options.loc === true,
         log: typeof options.loggerFn === 'function'
@@ -79,15 +80,15 @@ function createParseSettings(code, options = {}) {
         log('parserOptions.programs was provided, so parserOptions.project will be ignored.');
     }
     // Providing a program overrides project resolution
-    if (!parseSettings.programs) {
-        parseSettings.projects = (0, resolveProjectList_1.resolveProjectList)({
-            cacheLifetime: options.cacheLifetime,
-            project: (0, getProjectConfigFiles_1.getProjectConfigFiles)(parseSettings, options.project),
-            projectFolderIgnoreList: options.projectFolderIgnoreList,
-            singleRun: parseSettings.singleRun,
-            tsconfigRootDir: tsconfigRootDir,
-        });
-    }
+    // if (!parseSettings.programs) {
+    //     parseSettings.projects = (0, resolveProjectList_1.resolveProjectList)({
+    //         cacheLifetime: options.cacheLifetime,
+    //         project: (0, getProjectConfigFiles_1.getProjectConfigFiles)(parseSettings, options.project),
+    //         projectFolderIgnoreList: options.projectFolderIgnoreList,
+    //         singleRun: parseSettings.singleRun,
+    //         tsconfigRootDir: tsconfigRootDir,
+    //     });
+    // }
     (0, warnAboutTSVersion_1.warnAboutTSVersion)(parseSettings);
     return parseSettings;
 }
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/getProjectConfigFiles.js b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/getProjectConfigFiles.js
index f0ec41a..c5fa9e2 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/getProjectConfigFiles.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/getProjectConfigFiles.js
@@ -28,8 +28,8 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.getProjectConfigFiles = void 0;
 const debug_1 = __importDefault(require("debug"));
-const fs = __importStar(require("fs"));
-const path = __importStar(require("path"));
+// const fs = __importStar(require("fs"));
+// const path = __importStar(require("path"));
 const log = (0, debug_1.default)('typescript-eslint:typescript-estree:getProjectConfigFiles');
 /**
  * Checks for a matching TSConfig to a file including its parent directories,
@@ -48,23 +48,23 @@ function getProjectConfigFiles(parseSettings, project) {
             : [project];
     }
     log('Looking for tsconfig.json at or above file: %s', parseSettings.filePath);
-    let directory = path.dirname(parseSettings.filePath);
-    const checkedDirectories = [directory];
-    do {
-        log('Checking tsconfig.json path: %s', directory);
-        const tsconfigPath = path.join(directory, 'tsconfig.json');
-        const cached = (_a = parseSettings.tsconfigMatchCache.get(directory)) !== null && _a !== void 0 ? _a : (fs.existsSync(tsconfigPath) && tsconfigPath);
-        if (cached) {
-            for (const directory of checkedDirectories) {
-                parseSettings.tsconfigMatchCache.set(directory, cached);
-            }
-            return [cached];
-        }
-        directory = path.dirname(directory);
-        checkedDirectories.push(directory);
-    } while (directory.length > 1 &&
-        directory.length >= parseSettings.tsconfigRootDir.length);
-    throw new Error(`project was set to \`true\` but couldn't find any tsconfig.json relative to '${parseSettings.filePath}' within '${parseSettings.tsconfigRootDir}'.`);
+    // let directory = path.dirname(parseSettings.filePath);
+    // const checkedDirectories = [directory];
+    // do {
+    //     log('Checking tsconfig.json path: %s', directory);
+    //     const tsconfigPath = path.join(directory, 'tsconfig.json');
+    //     const cached = (_a = parseSettings.tsconfigMatchCache.get(directory)) !== null && _a !== void 0 ? _a : (fs.existsSync(tsconfigPath) && tsconfigPath);
+    //     if (cached) {
+    //         for (const directory of checkedDirectories) {
+    //             parseSettings.tsconfigMatchCache.set(directory, cached);
+    //         }
+    //         return [cached];
+    //     }
+    //     directory = path.dirname(directory);
+    //     checkedDirectories.push(directory);
+    // } while (directory.length > 1 &&
+    //     directory.length >= parseSettings.tsconfigRootDir.length);
+    // throw new Error(`project was set to \`true\` but couldn't find any tsconfig.json relative to '${parseSettings.filePath}' within '${parseSettings.tsconfigRootDir}'.`);
 }
 exports.getProjectConfigFiles = getProjectConfigFiles;
 //# sourceMappingURL=getProjectConfigFiles.js.map
\ No newline at end of file
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/inferSingleRun.js b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/inferSingleRun.js
index 6957a5c..3e5909b 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/inferSingleRun.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/inferSingleRun.js
@@ -1,7 +1,7 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.inferSingleRun = void 0;
-const path_1 = require("path");
+// const path_1 = require("path");
 /**
  * ESLint (and therefore typescript-eslint) is used in both "single run"/one-time contexts,
  * such as an ESLint CLI invocation, and long-running sessions (such as continuous feedback
@@ -31,15 +31,15 @@ function inferSingleRun(options) {
         return true;
     }
     // Currently behind a flag while we gather real-world feedback
-    if (options === null || options === void 0 ? void 0 : options.allowAutomaticSingleRunInference) {
-        if (
-        // Default to single runs for CI processes. CI=true is set by most CI providers by default.
-        process.env.CI === 'true' ||
-            // This will be true for invocations such as `npx eslint ...` and `./node_modules/.bin/eslint ...`
-            process.argv[1].endsWith((0, path_1.normalize)('node_modules/.bin/eslint'))) {
-            return true;
-        }
-    }
+    // if (options === null || options === void 0 ? void 0 : options.allowAutomaticSingleRunInference) {
+    //     if (
+    //     // Default to single runs for CI processes. CI=true is set by most CI providers by default.
+    //     process.env.CI === 'true' ||
+    //         // This will be true for invocations such as `npx eslint ...` and `./node_modules/.bin/eslint ...`
+    //         process.argv[1].endsWith((0, path_1.normalize)('node_modules/.bin/eslint'))) {
+    //         return true;
+    //     }
+    // }
     /**
      * We default to assuming that this run could be part of a long-running session (e.g. in an IDE)
      * and watch programs will therefore be required.
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.d.ts b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.d.ts
deleted file mode 100644
index 3db071e..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.d.ts
+++ /dev/null
@@ -1,19 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.d.ts.map b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.d.ts.map
deleted file mode 100644
index a248d3b..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.d.ts.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.js b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.js
deleted file mode 100644
index 59b273d..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.js
+++ /dev/null
@@ -1,103 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.js.map b/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.js.map
deleted file mode 100644
index c9043fa..0000000
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parseSettings/resolveProjectList.js.map
+++ /dev/null
@@ -1 +0,0 @@
diff --git a/node_modules/@typescript-eslint/typescript-estree/dist/parser.js b/node_modules/@typescript-eslint/typescript-estree/dist/parser.js
index ca184bd..dda2a2e 100644
--- a/node_modules/@typescript-eslint/typescript-estree/dist/parser.js
+++ b/node_modules/@typescript-eslint/typescript-estree/dist/parser.js
@@ -7,9 +7,9 @@ exports.clearParseAndGenerateServicesCalls = exports.clearProgramCache = exports
 const debug_1 = __importDefault(require("debug"));
 const ast_converter_1 = require("./ast-converter");
 const convert_1 = require("./convert");
-const createDefaultProgram_1 = require("./create-program/createDefaultProgram");
+// const createDefaultProgram_1 = require("./create-program/createDefaultProgram");
 const createIsolatedProgram_1 = require("./create-program/createIsolatedProgram");
-const createProjectProgram_1 = require("./create-program/createProjectProgram");
+// const createProjectProgram_1 = require("./create-program/createProjectProgram");
 const createSourceFile_1 = require("./create-program/createSourceFile");
 const useProvidedPrograms_1 = require("./create-program/useProvidedPrograms");
 const createParseSettings_1 = require("./parseSettings/createParseSettings");
@@ -33,10 +33,10 @@ exports.clearProgramCache = clearProgramCache;
 function getProgramAndAST(parseSettings, shouldProvideParserServices) {
     return ((parseSettings.programs &&
         (0, useProvidedPrograms_1.useProvidedPrograms)(parseSettings.programs, parseSettings)) ||
-        (shouldProvideParserServices && (0, createProjectProgram_1.createProjectProgram)(parseSettings)) ||
-        (shouldProvideParserServices &&
-            parseSettings.createDefaultProgram &&
-            (0, createDefaultProgram_1.createDefaultProgram)(parseSettings)) ||
+        // (shouldProvideParserServices && (0, createProjectProgram_1.createProjectProgram)(parseSettings)) ||
+        // (shouldProvideParserServices &&
+        //     parseSettings.createDefaultProgram &&
+        //     (0, createDefaultProgram_1.createDefaultProgram)(parseSettings)) ||
         (0, createIsolatedProgram_1.createIsolatedProgram)(parseSettings));
 }
 function parse(code, options) {
```

**Comment by bradzacher** [comment]
TBH I think that patching lime this is not good because you've fundamentally changed the source code for the tool.
This patch is going to be difficult to maintain as we make changes to our tooling, and it's impossible to validate that it introduces no behaviour change to the tooling either.

IMO a better option is doing something like what we do for our website - create stub mocks for the necessary nodejs packages so you can control how things work.

Regardless, closing as there is no action for our project.
